dask/dataframe/io/io.py

```diff
@@ -750,5 +754,145 @@ def sorted_division_locations(seq, npartitions=None, chunksize=None):
     return values, positions
 
 
+@insert_meta_param_description
+def from_map(
+    func,
+    inputs,
```

Comment:
`map_partitions` takes it's inputs as `*args` rather than as an iterable of arguments (so does `blockwise`, though that's more private). Here we have a bit of an odd mixture of `inputs` for the `func` args, but then `**kwargs` down below for the keyword args passed to the `func`.

In an ideal world, I'd probably prefer something like this, but with an `input_args` tuple and an `input_kwargs` dict, since mixing kwargs that get picked off by `from_map` and kwargs that get passed to `func` is a common foot-gun that I run into. I suspect other users run into that as well.

But right now, I might prefer `inputs` to be `*inputs`, if only for consistency with how `map_partitions`, `bag.map` and `Client.map` work.

Reply:
Maybe worth mentioning that the inputs can also be `BlockwiseDep`s.

Reply:
>I do think that Client.map might still be a good API comparison, where it's unpacked as a list-of-iterables, with the constraint that each of the iterables must be the same length (or a BlockwiseDep with that length)

Okay, I see. That does makes sense to me. However, handling multiple iterable/BlockwiseDep inputs **would** require us to modify the `DataFrameIOLayer` API (which I believe is already being used outside Dask)

Reply:
I suppose we could just zip multiple iterables together (into the same BlockwiseDep)

Reply:
> I suppose we could just zip multiple iterables together (into the same BlockwiseDep)

Yeah, that's something along the lines of what I was thinking. I think that would be closer to how `map_partitions` behaves as well, where each element in the iterables in `from_map` more-or-less corresponds to a single partition. I'm willing to be talked out of it, though, just aiming for more consistency.